./configure: code generation, TypeError: 'type' object is not subscriptable

After installing gcc11 and SDL2 before, I'm stuck with the below problem.
I find it hard to debug the problem. Did I invoke wrongly?
How do I get more information from the build invokation, apart from the logs?

Thanks for any pointers :) 

```
$ ./configure --download-nyan --mode debug --sanitize yes --sanitize-fatal .
./configure is a convenience script:
it creates the build directory,  symlinks it,
and invokes cmake for an out-of-source build.

Nobody is stopping you from skipping ./configure and our Makefile,
and using CMake directly (e.g. when packaging, or using an IDE).
For your convenience, ./configure even prints the direct CMake invocation!

         build_type | Debug
       cxx_compiler | g++
          cxx_flags | 
   exe_linker_flags | 
     install_prefix | /usr/local
module_linker_flags | 
shared_linker_flags | 

config options:

          backtrace | if_available
gperftools-profiler | if_available
gperftools-tcmalloc | False
            inotify | if_available
            ncurses | if_available
             opengl | if_available
             vulkan | if_available

bindir:
/home/chris/Code/openage/.bin/g++-debug-Oauto-sanitize-none/

invocation:
cmake -DCMAKE_BUILD_TYPE=Debug -DCMAKE_CXX_COMPILER=g++ -DCMAKE_CXX_FLAGS='' -DCMAKE_EXE_LINKER_FLAGS='' -DCMAKE_INSTALL_PREFIX=/usr/local -DCMAKE_MODULE_LINKER_FLAGS='' -DCMAKE_SHARED_LINKER_FLAGS='' -DCXX_OPTIMIZATION_LEVEL=auto -DCXX_SANITIZE_FATAL=False -DCXX_SANITIZE_MODE=none -DWANT_BACKTRACE=if_available -DWANT_GPERFTOOLS_PROFILER=if_available -DWANT_GPERFTOOLS_TCMALLOC=False -DWANT_INOTIFY=if_available -DWANT_NCURSES=if_available -DWANT_OPENGL=if_available -DWANT_VULKAN=if_available -DSDL2IMAGE_INCLUDE_DIRS=/usr/local/lib --download-nyan --mode debug --sanitize yes --sanitize-fatal . /home/chris/Code/openage

(now running cmake:)



 ___  ______ _______ _______ ___
|  _)/ _____|_______|_______|_  |
| | ( (____  _____      _     | |    ___  ____  _____ ____  _____  ____ _____
| |  \____ \|  ___)    | |    | |   / _ \|  _ \| ___ |  _ \(____ |/ _  | ___ |
| |_ _____) ) |        | |   _| |  | |_| | |_| | ____| | | / ___ ( (_| | ____|
|___|______/|_|        |_|  (___|   \___/|  __/|_____)_| |_\_____|\___ |_____)
                                         |_|                     (_____|

Welcome to the SFT technologies computer-aided openage build system!

You have chosen, or been chosen, to attempt the daring task of building openage.
If you have installed all the dependencies that are conveniently listed in
[doc/building.md], this _might_ just work!

If it doesn't, consider reporting the issue: https://github.com/SFTtech/openage
Or ask for help:
  * Matrix: #sfttech:matrix.org
  * IRC:    #sfttech on libera.chat

-- The CXX compiler identification is GNU 11.1.0
-- Check for working CXX compiler: /usr/bin/g++
-- Check for working CXX compiler: /usr/bin/g++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
========================================================
-- Performing Test HAVE_GSPLIT_DWARF_SUPPORT
-- Performing Test HAVE_GSPLIT_DWARF_SUPPORT - Success
-- Performing Test HAVE_REQUIRED_CXX20_SUPPORT
-- Performing Test HAVE_REQUIRED_CXX20_SUPPORT - Success
-- Found Python3: /usr/bin/python3.8 (found suitable exact version "3.8.10") found components: Interpreter Development NumPy 
-- Found Python: /usr/bin/python3.8 (Required is at least version "3.9") 
-- Found Cython: /usr/bin/python3.8 -m cython (found suitable version "0.29.14", minimum required is "0.25") 
-- Checking python3 module PIL.Image - Success
-- Checking python3 module PIL.ImageDraw - Success
-- Checking python3 module numpy - Success
-- Checking python3 module pygments - Success
-- Checking python3 module jinja2 - Success
-- Set PROJECT_VERSION from git.
-- Found Doxygen: /usr/bin/doxygen (found version "1.8.17") found components: doxygen dot 
-- The C compiler identification is GNU 11.1.0
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Found Freetype: /usr/lib/x86_64-linux-gnu/libfreetype.so (found version "2.10.1") 
-- Found ZLIB: /usr/lib/x86_64-linux-gnu/libz.so (found version "1.2.11") 
-- Found PNG: /usr/lib/x86_64-linux-gnu/libpng.so (found version "1.6.37") 
-- Found SDL2Image: /usr/lib/x86_64-linux-gnu/libSDL2_image.so  
-- Found Opusfile: /usr/lib/libopusfile.so  
-- Found Epoxy: /usr/lib/x86_64-linux-gnu/libepoxy.so  
-- Found HarfBuzz: /usr/lib/x86_64-linux-gnu/libharfbuzz.so (found suitable version "2.6.4", minimum required is "1.0.0") 
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Failed
-- Looking for pthread_create in pthreads
-- Looking for pthread_create in pthreads - not found
-- Looking for pthread_create in pthread
-- Looking for pthread_create in pthread - found
-- Found Threads: TRUE  
-- Could NOT find GCCBacktrace (missing: GCCBacktrace_LIBRARIES GCCBacktrace_INCLUDE_DIRS) 
-- optional dependency is unavailable: backtrace
-- Could NOT find GPerfTools (missing: GPERFTOOLS_LIBRARIES GPERFTOOLS_INCLUDE_DIR) 
-- optional dependency is unavailable: gperftools-profiler
-- Found Inotify: /usr/include/x86_64-linux-gnu  
-- Looking for wsyncup in /usr/lib/x86_64-linux-gnu/libcurses.so
-- Looking for wsyncup in /usr/lib/x86_64-linux-gnu/libcurses.so - found
-- Found Curses: /usr/lib/x86_64-linux-gnu/libncursesw.so  
-- Found OpenGL: /usr/lib/x86_64-linux-gnu/libOpenGL.so   
-- Found Vulkan: /usr/lib/x86_64-linux-gnu/libvulkan.so  

cython module
	run                                                 [embedded interpreter] [noinstall]
	openage.cython_check
	openage.cabextract.lzxd
	openage.cabextract.cabchecksum                      [standalone]
	openage.convert.processor.export.terrain_merge
	openage.convert.processor.export.texture_merge
	openage.convert.service.export.interface.visgrep
-- Found Ogg: /usr/lib/x86_64-linux-gnu/libogg.so  
	openage.convert.service.export.opus.opusenc
	openage.convert.service.export.png.binpack
	openage.convert.service.export.png.png_create
	openage.convert.value_object.read.media.slp
	openage.convert.value_object.read.media.smp
	openage.convert.value_object.read.media.smx
	openage.cppinterface.exctranslate
	openage.cppinterface.exctranslate_tests
	openage.cppinterface.pyobject
	openage.cppinterface.setup_checker
	openage.cvar.cvar
	openage.event.demo
	openage.game.main_cpp
	openage.log.log_cpp
	openage.main.main_cpp
	openage.main.tests
	openage.renderer.renderer_cpp
	openage.renderer.tests
	openage.testing.cpp_testing
	openage.testing.misc_cpp
	openage.util.filelike.cpp
	openage.util.fslike.cpp
	openage.versions.versions
Traceback (most recent call last):
  File "/usr/lib/python3.8/runpy.py", line 194, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/usr/lib/python3.8/runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "/home/chris/Code/openage/.bin/g++-debug-Oauto-sanitize-none/openage/__main__.py", line 172, in <module>
    sys.exit(main())
  File "/home/chris/Code/openage/.bin/g++-debug-Oauto-sanitize-none/openage/__main__.py", line 105, in main
    from .game.main import init_subparser
  File "/home/chris/Code/openage/.bin/g++-debug-Oauto-sanitize-none/openage/game/main.py", line 13, in <module>
    from ..convert.tool.subtool.acquire_sourcedir import wanna_convert
  File "/home/chris/Code/openage/.bin/g++-debug-Oauto-sanitize-none/openage/convert/tool/subtool/acquire_sourcedir.py", line 120, in <module>
    def query_source_dir(proposals: set[str]) -> AnyStr:
TypeError: 'type' object is not subscriptable
CMake Error at buildsystem/codegen.cmake:32 (message):
  failed to get target list from codegen invocation
Call Stack (most recent call first):
  buildsystem/python.cmake:494 (codegen_run)
  CMakeLists.txt:258 (python_finalize)
```

Mh tried with a 3.9 local virtualenv, but then then it fails because its not 3.8 exact version.

```
-- Could NOT find Python3: Found unsuitable version "3.9.5", but required is exact version "3.8" (found /home/chris/Code/openage/.venv/bin/python3)
CMake Error at /home/chris/Code/openage/.bin/g++-debug-Oauto-sanitize-none/CMakeFiles/CMakeTmp/CMakeLists.txt:17 (add_executable):
  Target "cmTC_25478" links to target "Python3::Python" but the target was
  not found.  Perhaps a find_package() call is missing for an IMPORTED
  target, or an ALIAS target is missing?
```



Full log
```
$ ./configure -- -DPYTHON_DIR="./.venv/bin" 
./configure is a convenience script:
it creates the build directory,  symlinks it,
and invokes cmake for an out-of-source build.

Nobody is stopping you from skipping ./configure and our Makefile,
and using CMake directly (e.g. when packaging, or using an IDE).
For your convenience, ./configure even prints the direct CMake invocation!

         build_type | Debug
       cxx_compiler | g++
          cxx_flags | 
   exe_linker_flags | 
     install_prefix | /usr/local
module_linker_flags | 
shared_linker_flags | 

config options:

          backtrace | if_available
gperftools-profiler | if_available
gperftools-tcmalloc | False
            inotify | if_available
            ncurses | if_available
             opengl | if_available
             vulkan | if_available

bindir:
/home/chris/Code/openage/.bin/g++-debug-Oauto-sanitize-none/

invocation:
cmake -DCMAKE_BUILD_TYPE=Debug -DCMAKE_CXX_COMPILER=g++ -DCMAKE_CXX_FLAGS='' -DCMAKE_EXE_LINKER_FLAGS='' -DCMAKE_INSTALL_PREFIX=/usr/local -DCMAKE_MODULE_LINKER_FLAGS='' -DCMAKE_SHARED_LINKER_FLAGS='' -DCXX_OPTIMIZATION_LEVEL=auto -DCXX_SANITIZE_FATAL=False -DCXX_SANITIZE_MODE=none -DWANT_BACKTRACE=if_available -DWANT_GPERFTOOLS_PROFILER=if_available -DWANT_GPERFTOOLS_TCMALLOC=False -DWANT_INOTIFY=if_available -DWANT_NCURSES=if_available -DWANT_OPENGL=if_available -DWANT_VULKAN=if_available -DPYTHON_DIR=./.venv/bin /home/chris/Code/openage

(now running cmake:)



 ___  ______ _______ _______ ___
|  _)/ _____|_______|_______|_  |
| | ( (____  _____      _     | |    ___  ____  _____ ____  _____  ____ _____
| |  \____ \|  ___)    | |    | |   / _ \|  _ \| ___ |  _ \(____ |/ _  | ___ |
| |_ _____) ) |        | |   _| |  | |_| | |_| | ____| | | / ___ ( (_| | ____|
|___|______/|_|        |_|  (___|   \___/|  __/|_____)_| |_\_____|\___ |_____)
                                         |_|                     (_____|

Welcome to the SFT technologies computer-aided openage build system!

You have chosen, or been chosen, to attempt the daring task of building openage.
If you have installed all the dependencies that are conveniently listed in
[doc/building.md], this _might_ just work!

If it doesn't, consider reporting the issue: https://github.com/SFTtech/openage
Or ask for help:
  * Matrix: #sfttech:matrix.org
  * IRC:    #sfttech on libera.chat

-- The CXX compiler identification is GNU 11.1.0
-- Check for working CXX compiler: /usr/bin/g++
-- Check for working CXX compiler: /usr/bin/g++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
========================================================
-- Performing Test HAVE_GSPLIT_DWARF_SUPPORT
-- Performing Test HAVE_GSPLIT_DWARF_SUPPORT - Success
-- Performing Test HAVE_REQUIRED_CXX20_SUPPORT
-- Performing Test HAVE_REQUIRED_CXX20_SUPPORT - Success
-- Could NOT find Python3: Found unsuitable version "3.9.5", but required is exact version "3.8" (found /home/chris/Code/openage/.venv/bin/python3)
CMake Error at /home/chris/Code/openage/.bin/g++-debug-Oauto-sanitize-none/CMakeFiles/CMakeTmp/CMakeLists.txt:17 (add_executable):
  Target "cmTC_25478" links to target "Python3::Python" but the target was
  not found.  Perhaps a find_package() call is missing for an IMPORTED
  target, or an ALIAS target is missing?


CMake Error at buildsystem/modules/FindPython.cmake:64 (try_compile):
  Failed to generate test project build system.
Call Stack (most recent call first):
  buildsystem/HandlePythonOptions.cmake:6 (find_package)
  CMakeLists.txt:175 (include)


-- Configuring incomplete, errors occurred!
See also "/home/chris/Code/openage/.bin/g++-debug-Oauto-sanitize-none/CMakeFiles/CMakeOutput.log".
```

Same as above
```
$ rm -rf build
$ source .venv/bin/activate
$ ./configure 
./configure is a convenience script:
it creates the build directory,  symlinks it,
and invokes cmake for an out-of-source build.

Nobody is stopping you from skipping ./configure and our Makefile,
and using CMake directly (e.g. when packaging, or using an IDE).
For your convenience, ./configure even prints the direct CMake invocation!

         build_type | Debug
       cxx_compiler | g++
          cxx_flags | 
   exe_linker_flags | 
     install_prefix | /usr/local
module_linker_flags | 
shared_linker_flags | 

config options:

          backtrace | if_available
gperftools-profiler | if_available
gperftools-tcmalloc | False
            inotify | if_available
            ncurses | if_available
             opengl | if_available
             vulkan | if_available

bindir:
/home/chris/Code/openage/.bin/g++-debug-Oauto-sanitize-none/

invocation:
cmake -DCMAKE_BUILD_TYPE=Debug -DCMAKE_CXX_COMPILER=g++ -DCMAKE_CXX_FLAGS='' -DCMAKE_EXE_LINKER_FLAGS='' -DCMAKE_INSTALL_PREFIX=/usr/local -DCMAKE_MODULE_LINKER_FLAGS='' -DCMAKE_SHARED_LINKER_FLAGS='' -DCXX_OPTIMIZATION_LEVEL=auto -DCXX_SANITIZE_FATAL=False -DCXX_SANITIZE_MODE=none -DWANT_BACKTRACE=if_available -DWANT_GPERFTOOLS_PROFILER=if_available -DWANT_GPERFTOOLS_TCMALLOC=False -DWANT_INOTIFY=if_available -DWANT_NCURSES=if_available -DWANT_OPENGL=if_available -DWANT_VULKAN=if_available /home/chris/Code/openage

(now running cmake:)



 ___  ______ _______ _______ ___
|  _)/ _____|_______|_______|_  |
| | ( (____  _____      _     | |    ___  ____  _____ ____  _____  ____ _____
| |  \____ \|  ___)    | |    | |   / _ \|  _ \| ___ |  _ \(____ |/ _  | ___ |
| |_ _____) ) |        | |   _| |  | |_| | |_| | ____| | | / ___ ( (_| | ____|
|___|______/|_|        |_|  (___|   \___/|  __/|_____)_| |_\_____|\___ |_____)
                                         |_|                     (_____|

Welcome to the SFT technologies computer-aided openage build system!

You have chosen, or been chosen, to attempt the daring task of building openage.
If you have installed all the dependencies that are conveniently listed in
[doc/building.md], this _might_ just work!

If it doesn't, consider reporting the issue: https://github.com/SFTtech/openage
Or ask for help:
  * Matrix: #sfttech:matrix.org
  * IRC:    #sfttech on libera.chat

-- The CXX compiler identification is GNU 11.1.0
-- Check for working CXX compiler: /usr/bin/g++
-- Check for working CXX compiler: /usr/bin/g++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
========================================================
-- Performing Test HAVE_GSPLIT_DWARF_SUPPORT
-- Performing Test HAVE_GSPLIT_DWARF_SUPPORT - Success
-- Performing Test HAVE_REQUIRED_CXX20_SUPPORT
-- Performing Test HAVE_REQUIRED_CXX20_SUPPORT - Success
-- Could NOT find Python3 (missing: Python3_LIBRARIES Python3_INCLUDE_DIRS Python3_NumPy_INCLUDE_DIRS Development NumPy) (found suitable exact version "3.9.5")
CMake Error at /home/chris/Code/openage/.bin/g++-debug-Oauto-sanitize-none/CMakeFiles/CMakeTmp/CMakeLists.txt:17 (add_executable):
  Target "cmTC_02814" links to target "Python3::Python" but the target was
  not found.  Perhaps a find_package() call is missing for an IMPORTED
  target, or an ALIAS target is missing?


CMake Error at buildsystem/modules/FindPython.cmake:64 (try_compile):
  Failed to generate test project build system.
Call Stack (most recent call first):
  buildsystem/HandlePythonOptions.cmake:6 (find_package)
  CMakeLists.txt:175 (include)


-- Configuring incomplete, errors occurred!
See also "/home/chris/Code/openage/.bin/g++-debug-Oauto-sanitize-none/CMakeFiles/CMakeOutput.log".
```

--- a/buildsystem/CheckRuntimeDependencies.cmake
+++ b/buildsystem/CheckRuntimeDependencies.cmake
@@ -1,11 +1,11 @@
-# Copyright 2015-2018 the openage authors. See copying.md for legal info.
+# Copyright 2015-2022 the openage authors. See copying.md for legal info.
 
 # python modules
 # a list of imported modules may be obtained via
 #
 # grep -RE '^ *(import |from [^.])' | cut -d: -f2- | \
 #     sed 's/^ *//g' | sort -u | grep -v openage
-set(REQUIRED_PYTHON_MODULES "PIL.Image" "PIL.ImageDraw" "numpy" "pygments" "jinja2")
+set(REQUIRED_PYTHON_MODULES "PIL.Image" "PIL.ImageDraw" "numpy" "pygments" "jinja2" "toml" "lz4")
 
 # command-line tools
 # example: set(REQUIRED_UTILITIES "foobar")

--- a/buildsystem/HandlePythonOptions.cmake
+++ b/buildsystem/HandlePythonOptions.cmake
@@ -1,4 +1,4 @@
-# Copyright 2015-2021 the openage authors. See copying.md for legal info.
+# Copyright 2015-2022 the openage authors. See copying.md for legal info.
 
 # finds the python interpreter, install destination and extension flags.
 
@@ -45,7 +45,7 @@ if(NOT CMAKE_PY_INSTALL_PREFIX)
 	if(MSVC)
 		set(CMAKE_PY_INSTALL_PREFIX "python")
 	else()
-		py_exec("from distutils.sysconfig import get_python_lib; print(get_python_lib(prefix='${CMAKE_INSTALL_PREFIX}'))" PREFIX)
+		py_exec("import sysconfig, os; print(os.path.join('${CMAKE_INSTALL_PREFIX}', os.path.relpath(sysconfig.get_path('purelib'), sysconfig.get_path('data'))))" PREFIX)
 		set(CMAKE_PY_INSTALL_PREFIX "${PREFIX}")
 	endif()
 endif()

--- a/buildsystem/modules/FindPython.cmake
+++ b/buildsystem/modules/FindPython.cmake
@@ -25,28 +25,39 @@
 # to CMake. It's used as a hint where to look at
 if(PYTHON_DIR)
 	set(Python3_ROOT_DIR "${PYTHON_DIR}")
+	set(PYTHON_USED_VERSION "${PYTHON_MIN_VERSION}")
+
+else()
+	# when there are multiple pythons, preferrably use the version of
+	# the default `python3` executable.
+	execute_process(
+		COMMAND "python3" -c "import platform; print(platform.python_version())"
+		OUTPUT_VARIABLE PYVER_OUTPUT
+		RESULT_VARIABLE PYVER_RETVAL
+	)
+
+	# if a system version exists, check if it's compatible with our min requirements
+	if(PYVER_RETVAL EQUAL 0)
+		string(REGEX MATCH "^[0-9]+\\.[0-9]+" PYTHON_USED_VERSION "${PYVER_OUTPUT}")
+
+		if(PYTHON_USED_VERSION VERSION_GREATER_EQUAL PYTHON_MIN_VERSION)
+			# set EXACT so we get the system version from find_package
+			set(need_exact_version "EXACT")
+
+		else()
+			# search for alternatives if version doesn't fulfill min requirements
+			set(PYTHON_USED_VERSION "${PYTHON_MIN_VERSION}")
+
+		endif()
+	endif()
 endif()
 ###############################################################
-
 
 # Never use the Windows Registry to find python
 set(Python3_FIND_REGISTRY "NEVER")
 
-# when there are multiple pythons, preferrably use the version of
-# the default `python3` executable.
-execute_process(
-	COMMAND "python3" -c "import platform; print(platform.python_version())"
-	OUTPUT_VARIABLE PYVER_OUTPUT
-	RESULT_VARIABLE PYVER_RETVAL
-)
-
-if(PYVER_RETVAL EQUAL 0)
-	string(REGEX MATCH "^[0-9]+\\.[0-9]+" PYTHON_MIN_VERSION "${PYVER_OUTPUT}")
-	set(need_exact_version "EXACT")
-endif()
-
 # use cmake's FindPython3 to locate library and interpreter
-find_package(Python3 ${PYTHON_MIN_VERSION} ${need_exact_version} COMPONENTS Interpreter Development NumPy)
+find_package(Python3 ${PYTHON_USED_VERSION} ${need_exact_version} COMPONENTS Interpreter Development NumPy)
 
 # python version string to cpython api test in modules/FindPython_test.cpp
 # we use the solution from CPython's header (see https://github.com/SFTtech/openage/issues/1438#issuecomment-1036311012):

--- a/buildsystem/modules/FindPython.cmake
+++ b/buildsystem/modules/FindPython.cmake
@@ -133,7 +133,7 @@ endfunction()
 function(py_get_config_var VAR RESULTVAR)
 	# uses py_exec to determine a config var as in distutils.sysconfig.get_config_var().
 	py_exec(
-		"from distutils.sysconfig import get_config_var; print(get_config_var('${VAR}'))"
+		"from sysconfig import get_config_var; print(get_config_var('${VAR}'))"
 		RESULT
 	)
 

--- a/doc/building.md
+++ b/doc/building.md
@@ -60,9 +60,13 @@ Dependency list:
       A   An installed version of any of the following (wine is your friend).
           Other versions _might_ work:
 
+     - Age of Empires I: Rise of Rome Patch 1.0a
+     - Age of Empires I: Definitive Edition
      - Age of Empires II: The Conquerors Patch 1.0c
      - Age of Empires II: Forgotten Empires
-     - Age of Empires II HD
+     - Age of Empires II: HD (now also called: Age of Empires II (2013))
+     - Age of Empires II: Definitive Edition
+     - Star Wars: Galactic Battlegrounds: Clone Campaigns
 
 
 ### Dependency installation

